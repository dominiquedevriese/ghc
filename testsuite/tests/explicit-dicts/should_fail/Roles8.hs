{-# LANGUAGE RoleAnnotations #-}
{-# LANGUAGE TypeApplications #-}
{-# LANGUAGE TypeFamilies #-}
module Roles8 where

import GHC.Base (Ord(..))

reverseOrd :: Ord a => Ord.Dict a
reverseOrd = undefined

data Set a
type role Set nominal

insert :: Ord a => a -> Set a -> Set a
insert = undefined

evilInsert :: (Ord a, a ~ b) => a -> b -> Set b -> Set b
evilInsert _ = insert


f8 = evilInsert ((reverseOrd @Int :: Ord a))
