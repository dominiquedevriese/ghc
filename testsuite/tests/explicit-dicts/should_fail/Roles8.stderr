
Roles8.hs:21:6: error:
    • Explicit dictionary application to: ‘Ord a’
      is not allowed because of instance incoherence:
      In the context ‘a ~ b’
        Type variable ‘a’ has role Nominal
    • In the expression: evilInsert ((reverseOrd @Int :: Ord a))
      In an equation for ‘f8’:
          f8 = evilInsert ((reverseOrd @Int :: Ord a))
